### PR TITLE
feat(compiler)!: Implement `Infinity` and `NaN` syntax

### PR DESCRIPTION
**Pull request by spotandjake**
This pr implement `Infinity` and `NaN` syntax as I was talking about in dms with Oscar.

Closes: #693

### CODE REVIEW

File: compiler/src/parsing/lexer.re [hunk 1]
```diff
@@ -108,7 +108,8 @@ let dec_float_integral = [%sedlex.regexp?
 let dec_float = [%sedlex.regexp?
   (dec_float_integral, dec_float_decimal, Opt(dec_float_exp)) |
   (dec_float_decimal_explicit, Opt(dec_float_exp)) |
-  (dec_float_integral, dec_float_exp)
+  (dec_float_integral, dec_float_exp) |
+  ("Infinity" | "NaN")
```
Comment (ospencer) [inline compiler/src/parsing/lexer.re:112]: Just to make this slightly cleaner, I'd make a `dec_float_alphabetic` that's just this, and then reference it here.

File: compiler/test/__snapshots__/basic_functionality.4d1501b9.0.snapshot [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        )
+        (i32.const 5)
+       )
+       (i32.store offset=4
+        (local.get $0)
+        (i32.const 2)
+       )
+       (f64.store offset=8
+        (local.get $0)
+        (f64.const nan:0x8000000000000)
```
Comment (ospencer) [inline compiler/test/__snapshots__/basic_functionality.4d1501b9.0.snapshot:45]: Can't argue with that.

File: compiler/test/__snapshots__/basic_functionality.6599e10c.0.snapshot [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        )
+        (i32.const 5)
+       )
+       (i32.store offset=4
+        (local.get $0)
+        (i32.const 2)
+       )
+       (f64.store offset=8
+        (local.get $0)
+        (f64.const -inf)
```
Comment (ospencer) [inline compiler/test/__snapshots__/basic_functionality.6599e10c.0.snapshot:45]: The wat syntax for these is pretty nice!

File: compiler/test/suites/basic_functionality.re [hunk 4]
```diff
@@ -27,6 +27,9 @@ describe("basic functionality", ({test, testSkip}) => {
   assertSnapshot("oct_neg", "-0o77");
   assertSnapshot("fals", "let x = false; x");
   assertSnapshot("tru", "let x = true; x");
+  assertSnapshot("inf", "let x = Infinity; x");
+  assertSnapshot("neginf", "let x = -Infinity; x");
```
Comment (ospencer) [inline compiler/test/suites/basic_functionality.re:31]:
```suggestion
  assertSnapshot("infinity_neg", "let x = -Infinity; x");
```

File: compiler/test/suites/basic_functionality.re [hunk 5]
```diff
@@ -27,6 +27,9 @@ describe("basic functionality", ({test, testSkip}) => {
   assertSnapshot("oct_neg", "-0o77");
   assertSnapshot("fals", "let x = false; x");
   assertSnapshot("tru", "let x = true; x");
+  assertSnapshot("inf", "let x = Infinity; x");
```
Comment (ospencer) [inline compiler/test/suites/basic_functionality.re:30]:
```suggestion
  assertSnapshot("infinity", "let x = Infinity; x");
```

**Review by ospencer — CHANGES_REQUESTED** [review]
Looking good! Can you make an issue on the lsp repo to add syntax highlighting for this?

**Comment by spotandjake** [comment]
Applied Those Changes

**Comment by ospencer** [comment]
You'll have to regen the snapshots.

**Comment by spotandjake** [comment]
> You'll have to regen the snapshots.

I did?

**Comment by ospencer** [comment]
After accepting the suggestions, since we changed the names of the tests.

**Comment by spotandjake** [comment]
> After accepting the suggestions, since we changed the names of the tests.

yah I did in my commit for it the test are updated

**Review by ospencer — APPROVED** [review]
Looks great!

**Comment by spotandjake** [comment]
Rebased on main, and the tests the tests seem to be passing now.

**Pull request merged**